### PR TITLE
 project structure refactor based on clean architecture

### PR DESCRIPTION
**Pull request by LautaroSanchezGl**
- Changed Project structure based on clean architecture
- Added genre and movie repository interface

### CODE REVIEW

**Review by LeonardoMolleker — APPROVED** [review]

File: lib/data/repository/genre_repository.dart [hunk 1]
```diff
@@ -1,16 +1,17 @@
 import 'dart:convert';
 
 import 'package:flutter/services.dart';
-import '../entities/genre.dart';
+import '../../domain/repository/genre_repository_interface.dart';
+import '../../domain/entity/genre.dart';
 
-class GenreRepository {
+class GenreRepository implements GenreRepositoryInterface {
```
Comment (matiasilveiro-gl) [inline lib/data/repository/genre_repository.dart]: Minor: Personally I'd choose one of the below naming conventions:

- `class GenreRepositoryImpl implements GenreRepository`
- `class GenreRepository implements IGenreRepository`
- `class LocalGenreRepository implements GenreRepository`

Thus avoiding the `Interface` ending. Personal choice! If you don't agree with this, mark this as resolved by yourself :)

**Review by matiasilveiro-gl — APPROVED** [review]
LGTM!

File: lib/data/repository/genre_repository.dart (hunk 1 above)
Comment (LautaroSanchezGl) [reply lib/data/repository/genre_repository.dart]: seen second approach quite often... so changed to  class GenreRepositoryImp implements IGenRepository
Thanks!

File: lib/presentation/view/movie_list_view.dart [hunk 2]
```diff
@@ -12,7 +12,7 @@ class MovieListView extends StatefulWidget {
 }
 
 class _MovieListViewState extends State<MovieListView> {
-  final MovieRepository movieRepository = MovieRepository();
+  final MovieRepositoryImp movieRepository = MovieRepositoryImp();
```
Comment (matiasilveiro-gl) [inline lib/presentation/view/movie_list_view.dart]: This should be `final IMovieRepository movieRepository = MovieRepositoryImp();`. 
The presentation layer relies on interfaces from the domain layer! 
(I know we're initializing this to an implementation from the data layer, but it's just now, we'll update that in a later PR)

File: lib/presentation/widget/genres_row.dart [hunk 3]
```diff
@@ -17,7 +17,7 @@ class Genres extends StatefulWidget {
 }
 
 class _GenresState extends State<Genres> {
-  final GenreRepository genreRepository = GenreRepository();
+  final GenreRepositoryImp genreRepository = GenreRepositoryImp();
```
Comment (matiasilveiro-gl) [inline lib/presentation/widget/genres_row.dart]: This should be `final IGenreRepository movieRepository = GenreRepositoryImp();`. 
The presentation layer relies on interfaces from the domain layer!

File: lib/presentation/view/movie_list_view.dart (hunk 2 above)
Comment (LautaroSanchezGl) [reply lib/presentation/view/movie_list_view.dart]: you are right! done!

File: lib/presentation/widget/genres_row.dart (hunk 3 above)
Comment (LautaroSanchezGl) [reply lib/presentation/widget/genres_row.dart]: done

**Pull request merged**